assignments/array-methods.js

```diff
@@ -54,30 +56,75 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
+// Convert each first name into all caps and log the result
 let allCaps = [];
+const drunkDirector = runners.map(function(currentValue) {
+  let capitalized = currentValue["first_name"].toUpperCase();
+  allCaps.push(capitalized);
+})
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
-// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+// The large shirts won't be available for the event due to an ordering issue.  
+// Get a list of runners with large sized shirts so they can choose a different size. 
+// Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
+let largeShirts = runners.filter((currentValue) => {
+  return currentValue["shirt_size"] === "L";
+})
+
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+let ticketPriceTotal = runners.reduce(function(totalDonations, currentValue){
+  return totalDonations + currentValue["donation"];
+}, 0)
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
-// Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
+// Now that you have used .forEach(), .map(), .filter(), and .reduce().  
+// I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  
+// Try to create and then solve 3 unique problems using one or many of the array methods listed above.
+
+// Problem 1: Alphabetize by last name.
+let alphabatized = runners.map(function(currentValue) {
+  return currentValue;
+ }).sort(function(a,b) {
+   if (a.last_name > b.last_name) {
+     return 1;
+   } else if (a.last_name < b.last_name) {
+     return -1
+   } else {
+     return 0;
+   }
+ });
+ console.log(alphabatized);
+
+
+// console.log(alphabatized);
+
+// // Problem 2: New array with donation, first & last name ONLY, and sorted by donation size.
+
+const byDonationSize = runners.map(runner => {return {donation:runner.donation, first_name:runner.first_name, last_name:runner.last_name}})
+.sort((a,b) => a["donation"] - b["donation"]).reverse();
+
+console.log(byDonationSize);
 
-// Problem 1
+// Problem 3 Average donation size
 
-// Problem 2
+let totalDonations = runners.reduce(function(totalDonations, currentValue){
+  return totalDonations + currentValue["donation"];
+}, 0);
```

Comment:
```suggestion
}, 0) / runners.length;
```
You can chain on to end the division since runners.reduce will return a number